### ISSUE TITLE
Question: Is there a way to trigger an expression on text

### ISSUE THREAD

**Issue by didikunz** [airbnb/lottie-web#2573]
I try to build a Lottie animation with a expression, that scales the text, based on the amount of text entered. That works in After Effects, when i change the source text (so the expression is ok). But when I set the text from JavaScript, using "TextLayer.updateDocumentData" The expression is not doing anything. It looks like it is not triggered. Is there a way to trigger Lottie to re-evaluate the expressions, after a dynamic update or any other trick that I missed? Thanks.

**Comment by bodymovin** [comment]
can you share a reduced example with your implementation that I can test?

**Comment by didikunz** [comment]
Yes, you find the lottie file [here](https://www.dropbox.com/s/6h3txiuahiinebb/YellowCard_simple.json?dl=0)and the AE file [here](https://www.dropbox.com/s/kvsswl3yvjfje7h/Football_Example_Simple.aep?dl=0). Thanks.

**Comment by didikunz** [comment]
I probably should add, that the wxpression is in the scaling of layer 4 and 8 respectively. The expression is: 

```
var layerWidth = thisLayer.sourceRectAtTime().width;
var percentage = 650 / layerWidth * 100;
transform.scale = [Math.min(100, percentage), 100];

```
While 650 (pixels) is the width that the text should maximally be. Can it be, that Javascript does not interprete it as beeing pixels?

**Comment by bodymovin** [comment]
can you share a working example with the code that updates the text?

**Comment by didikunz** [comment]
It is not so easy to build a standalone HTML file, that shows the effect, I am currently on vaccation with a camp car and also does not have a webserver to deploy the example. But what I was able to try fiddling the code of my CasparCG template. The formula posted earlier is translated by BodyMovin to: 

`"x": "var $bm_rt;\nvar layerWidth = thisLayer.sourceRectAtTime().width;\nlog(layerWidth);\nvar percentage = $bm_mul($bm_div(650, layerWidth), 100);\n$bm_rt = transform.scale = [\n    Math.min(100, percentage),\n    100\n];"`

To me the line `$bm_rt = transform.scale = [\n    Math.min(100, percentage),\n    100\n];` looks wrong. How does it set the transformation?

**Comment by didikunz** [comment]
So I finally was able to create an example. You find it [here](https://mediasupport.ch/LottieTest/LottieTest.html). Bellow the (big) gray area is a text-field and a button. If you click the button, you see, that the expression does not process the text.

**Comment by didikunz** [comment]
@bodymovin any news on this?
